docs/src/createEmotionCache.ts

```diff
@@ -8,8 +8,8 @@ function globalSelector(element: Element) {
   switch (element.type) {
     case RULESET:
       element.props = (element.props as string[]).map((value: any) => {
-        if (value.match(/(:where|:is)\(/)) {
-          value = value.replace(/\.[^:]+(:where|:is)/, '$1');
+        if (value.match(/(:where|:has)\(/)) {
+          value = value.replace(/\.[^:]+(:where|:has)/, '$1');
```

Comment:
I am not sure what this code actually does, but after some digging, it seems that it removes the className attached in the beginning for `:where` and `:is` selectors. Emotion does not remove the className. See the issue reported https://github.com/emotion-js/emotion/issues/2836. 

The style shown in the image below was getting applied:

![Nav Bar regression screenshot](https://github.com/mui/material-ui/assets/20900032/cc274d04-fc55-49a3-adef-b1ce7bd6c3d8)

Because of:

https://github.com/mui/material-ui/blob/092b06a05320a17e55ca00e691956dacc3061262/docs/src/modules/brandingTheme.ts#L781-L784
which is in the branding theme.

This was changed in https://github.com/mui/material-ui/pull/37120/commits/3f6b77eeef7ce7ee3961e2af1b4e560021ff65ea in #37120. 


The Base UI's MultiSelect demo does not get affected by changing `:is` to `:has`.

cc @siriwatknp @michaldudak

Reply:
@oliviertassinari I cannot reproduce the "After" issue. I use Windows OS, Chrome browser `Version 113.0.5672.127 (Official Build) (64-bit)`.

---

Regarding your options:

1. We cannot ignore `:is` if it is used in future to target it globally anywhere. We can forget to add the universal selector (*) before it. This workaround will simply strip the className.
2. Fixed the syntax in https://github.com/mui/material-ui/pull/37362/commits/f348346de8d47f7192eeb8258539994877e9930a. See https://github.com/mui/material-ui/pull/37362#discussion_r1201824121.
3. `:where` is used in multiple places. IMO, it is nice to strip the className in one place without manipulating the syntax.

Reply:
It looks like the fix we did looks good.